systems/controllers/osc/operational_space_control.cc

```diff
@@ -0,0 +1,619 @@
+  VectorXd JdotV_h = tree_wo_spr_.positionConstraintsJacDotTimesV(cache_wo_spr);
+
+  // Get J and JdotV for contact constraint
+  MatrixXd J_c = MatrixXd::Zero(n_c_, n_v_);
+  VectorXd JdotV_c = VectorXd::Zero(n_c_);
+  for (unsigned int i = 0; i < active_contact_flags.size(); i++) {
+    if (active_contact_flags[i]) {
+      J_c.block(3 * i, 0, 3, n_v_) = tree_wo_spr_.transformPointsJacobian(
+                                       cache_wo_spr, pt_on_body_[i],
+                                       body_index_[i], 0, false);
+      JdotV_c.segment(3 * i, 3) = tree_wo_spr_.transformPointsJacobianDotTimesV(
+                                    cache_wo_spr, pt_on_body_[i],
+                                    body_index_[i], 0);
+    }
+  }
+
+  // Update constraints
+  // 1. Dynamics constraint
+  ///    M*dv + bias == J_c^T*lambda_c + J_h^T*lambda_h + B*u
+  /// -> M*dv - J_c^T*lambda_c - J_h^T*lambda_h - B*u == - bias
+  /// -> [M, -J_c^T, -J_h^T, -B]*[dv, lambda_c, lambda_h, u]^T = - bias
+  MatrixXd A_dyn = MatrixXd::Zero(n_v_, n_v_ + n_c_ + n_h_ + n_u_);
+  A_dyn.block(0, 0, n_v_, n_v_) = M;
+  A_dyn.block(0, n_v_, n_v_, n_c_) = -J_c.transpose();
+  A_dyn.block(0, n_v_ + n_c_ , n_v_, n_h_) = -J_h.transpose();
+  A_dyn.block(0, n_v_ + n_c_ + n_h_, n_v_, n_u_) = -B;
+  dynamics_constraint_->UpdateCoefficients(A_dyn, -bias);
+  // 2. Holonomic constraint
+  ///    JdotV_h + J_h*dv == 0
+  /// -> J_h*dv == -JdotV_h
+  holonomic_constraint_->UpdateCoefficients(J_h, -JdotV_h);
+  // 3. Contact constraint
+  if (w_soft_constraint_ <= 0) {
+    ///    JdotV_c + J_c*dv == 0
+    /// -> J_c*dv == -JdotV_c
+    contact_constraints_->UpdateCoefficients(J_c, -JdotV_c);
+  }
+  else {
+    // Relaxed version:
+    ///    JdotV_c + J_c*dv == -epsilon
+    /// -> J_c*dv + I*epsilon == -JdotV_c
+    /// -> [J_c, I]* [dv, epsilon]^T == -JdotV_c
+    MatrixXd A_c = MatrixXd::Zero(n_c_, n_v_ + n_c_);
+    A_c.block(0, 0, n_c_, n_v_) = J_c;
+    A_c.block(0, n_v_, n_c_, n_c_) = MatrixXd::Identity(n_c_, n_c_);
+    contact_constraints_->UpdateCoefficients(A_c, -JdotV_c);
+  }
+  // 4. Friction constraint (approximated firction cone)
+  /// For i = active contact indices
+  ///     mu_*lambda_c(3*i+2) >= lambda_c(3*i+0)
+  ///    -mu_*lambda_c(3*i+2) <= lambda_c(3*i+0)
+  ///     mu_*lambda_c(3*i+2) >= lambda_c(3*i+1)
+  ///    -mu_*lambda_c(3*i+2) <= lambda_c(3*i+1)
+  ///         lambda_c(3*i+2) >= 0
+  /// ->
+  ///     mu_*lambda_c(3*i+2) - lambda_c(3*i+0) >= 0
+  ///     mu_*lambda_c(3*i+2) + lambda_c(3*i+0) >= 0
+  ///     mu_*lambda_c(3*i+2) - lambda_c(3*i+1) >= 0
+  ///     mu_*lambda_c(3*i+2) + lambda_c(3*i+1) >= 0
+  ///                           lambda_c(3*i+2) >= 0
+  if (body_index_.size() > 0) {
+    VectorXd mu_minus1(2); mu_minus1 << mu_, -1;
+    VectorXd mu_plus1(2); mu_plus1 << mu_, 1;
+    VectorXd inf_vectorxd(1); inf_vectorxd << numeric_limits<double>::infinity();
+    for (unsigned int i = 0; i < active_contact_flags.size(); i++) {
```

Comment:
Probably `inf`--but better yet would be an efficient way to just remove the constraint.

Reply:
I can't find a way to remove a constraint. 
I wonder if the total runtime is short if we remove and add the constraint compared to only updating the coefficients?

Reply:
Leave it as a TODO for now.